Phonebook.Backend/Phonebook.Backend.PictureService/Phonebook.Backend.PictureService/BackgroundTasks/PurgeTask.cs

```diff
@@ -0,0 +1,117 @@
+﻿using KK.AspNetCore.BackgroundTasks.Scheduled;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Phonebook.Backend.PictureService.Configuration;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Phonebook.Backend.PictureService.Controllers
+{
+    public class PurgeTask : IScheduledTask
+    {
+
+        private readonly ILogger<PurgeTask> logger;
+        private readonly PictureServiceConfiguration configuration;
+
+        public PurgeTask(
+            IScheduledTaskOptions<PurgeTask> options,
+            ILogger<PurgeTask> logger,
+            PictureServiceConfiguration configuration
+            )
+        {
+            this.Options = options;
+            this.logger = logger;
+            this.configuration = configuration;
+        }
+
+        public IScheduledTaskOptions<IScheduledTask> Options { get; }
+
+        public async Task ExecuteAsync(CancellationToken stoppingToken)

```

Comment:
This function is way to complex! Please rewrite this. or wait some time so I can contribute

Reply:
Done.